### DISCUSSION TITLE
BBS02

### DISCUSSION THREAD

**Discussion by CitizenDev** [danielnilsson9/bbs-fw#48] [General]
Hi.

I created this thread in order to discuss anything related to BBS02 and maybe get some updates on development from time to time.

Just sent you second donation so it would reach needed 100$ :)
Thanks again for everything and I have my fingers crossed for BBS02 firmware.

**Comment by danielnilsson9** [comment]
Sounds good, thank you! I have ordered a controller now.

**Comment by gullygossner** [comment]
I'm excited for development! I was really sad to find out that the firmware of my bbs02 was locked down. Previously owning a bbshd, I miss being able to customize the parameters as the stock configuration is not very good in my opinion.

**Comment by danielnilsson9** [comment]
BBS02 is ready for early testing in the BBS02 branch. I have been running it on my bench and all functions seems to be working. You will have to build from source, both the firmware and config tool.

CPU frequency had to be changed to 20Mhz for technical reasons, make sure you select it correctly when you flash it with STC-ISP.

Tested on BBS02B controller, I have no idea if it would work with the first version... If you have STC15F2K60S2 when you run check mcu it should be fine.

NOTE: BBS02 can be set to run up to 33A too but I would not recommend that...

**Reply by gullygossner** [reply]
Great work! Am I missing something though, I am not seeing the BBS02 branch?

**Reply by danielnilsson9** [reply]
https://github.com/danielnilsson9/bbshd-fw/tree/bbs02

**Reply by gullygossner** [reply]
Great! I was able to get the config tool built and will try to build the firmware this weekend sometime to give it a shot.

**Comment by CitizenDev** [comment]
I still haven't managed to get original 48V BBS02 Firmware. There is 52V floating around but I'm running 48V myself. So for now I'm a bit hesitant testing this so early. 
Btw what was your BBS02 controller MCU type and F/W version?

**Reply by gullygossner** [reply]
I don't want to give bad advice but from what I have gathered the 52v hex will play nice with 48v batteries. The big downside to any of the hex's in the wild is that they are locked down so you can't modify your parameters once you've flashed it.

**Reply by gullygossner** [reply]
The low voltage cut off is set at 43V though so that may be a consideration.

**Reply by danielnilsson9** [reply]
Oh damn, I didn't know that the 52V hex was locked down. That is bad, then there is no way to go back to an unlocked original firmware if you flash this. 

I'm not planning any more development for version 1.1 at the moment, the last changes were pushed recently. If there are no issues found with the BBS02 version it will be included in the 1.1 release next week hopefully.

I do need someone to take the chance and flash this and test it out to confirm BBS02 version has no issues as I can't do it myself without a motor.

MCU is STC15F2K60S2, I don't remember the original firmware revision, was to fast to flash my own on it... There are no revision written anywhere, I don't think there are different version of the BBS02B controller?

**Reply by gullygossner** [reply]
The locked down hex was my main excitement for this firmware development. If I can find time to figure out how to compile the firmware, I'll flash it this weekend as my firmware is already locked down so I don't have anything to lose on that front.

**Reply by danielnilsson9** [reply]
I tried flashing this original firmware on my BBS02 controller, it says it is a 48V firmware when I later connect with BafangConfigTool and it seems to be possible to configure all settings.

https://support.biktrix.com/kb/guide/en/52v-programming-for-bbs02-motor-K2gUq2nWAX/Steps/1132765

I know the article says 52V but it probably just means that it will also accept 52V without throwing an error.

Manufacturer: HZXT
Model: SZZ9
Hardware Version: 1.1
Firmware Version : V2.0.1.1
Nominal Voltage: 48V
Max Current: 25A


So if you haven't tried my firmware yet I think you are safe since there is a working original firmware.

**Reply by CitizenDev** [reply]
This is fantastic news Daniel! I will probably reflash later tonight after work.

Thanks again.

**Comment by gullygossner** [comment]
Just so I am clear, I flash the main.hex file from the make compile and set STC ISP Programming Software Input IRC Frequency to 20MHz?

**Reply by danielnilsson9** [reply]
Yep! That is correct, you might want to update again since I found two small bugs I just fixed.

**Reply by gullygossner** [reply]
Unfortunately when I check mcu I am getting  MCU type: IAP15F2K61S2 F/W version: 7.2.5S so I don't think I should attempt to flash.

**Reply by danielnilsson9** [reply]
Interesting, sound like the same deal as for BBSHD then, they have switch between STC and IAP MCU, only difference is flash layout I suspect (will have to double check that). What I did for BBSHD was to implement auto detection at runtime to support both (only matters when saving config to flash).

Have you flashed the 52V original version and knows that it works?

**Reply by gullygossner** [reply]
I have flashed two 52v hex files. One from https://biktrix.stonly.com/kb/guide/en/52v-programming-for-bbs02-motor-K2gUq2nWAX/Steps/1132765 and one emailed to me from golden motor (currently using). Both work but as i mentioned both are locked down.

**Reply by danielnilsson9** [reply]
I have checked the documents, the flash layout difference is identical to the difference between BBSHD IAP vs STC MCU, the auto detection will work and solve that. I think you are safe to try to flash it and since you have already tried the other firmwares you can always go back if it doesn't work for some reason.

I have flashed that 52V firmware on mine and it worked so I do not expect you to have any issues.

**Reply by gullygossner** [reply]
Successfully flashed and configered. At first glance everything appears to be working.

**Reply by CitizenDev** [reply]
Let us know if you have driven your bike and all is OK. I'm next in line then :)

**Reply by gullygossner** [reply]
I took it out for about 4km yesterday and it worked fine.

**Reply by danielnilsson9** [reply]
It was discovered I had made an error which will give you symptoms such as small interruptions in power when using PAS and wrong speed indicated on display for short periods (both of which could be hard to notice). I have pushed  the changes to the bbs02 branch too. But the bbs02 branch will be removed at some point since it has now been merged to master. You can also build from master, all you have to do is select BBS02 as target controller in the makefile:
https://github.com/danielnilsson9/bbshd-fw/blob/master/src/firmware/makefile#L14

**Comment by CitizenDev** [comment]
So I built and flashed firmware successfully but Config tool fails to connect with controller after that.
When I connect the display and turn it on, it seems to work and communicate with the controller.
I tried both the release version of the tool and also tried to build from source in case there were changes after BBS02 migration.

**Reply by danielnilsson9** [reply]
Please build everything from master, I just pushed an important change.
What serial adapter model do you have, CH340?
Did you remember to set the CPU frequency to 20MHz?

**Reply by CitizenDev** [reply]
I did, with SDCC 4.0 about an hour ago, but if this change can be related I'll do another one.
Yes, CH340 according to config tool com port scan

**Reply by danielnilsson9** [reply]
Alright, no you should not be affected then, that's weird, did you get any error when trying to connect or is it just spinning?

**Reply by CitizenDev** [reply]
No error, eventually just timeout.

Check MCU on STC-ISP reads mcu type just fine

**Reply by danielnilsson9** [reply]
And about the CPU frequency set?

**Reply by CitizenDev** [reply]
It reads IRC freq. as 18.401 if that's what you meant

**Reply by danielnilsson9** [reply]
Yes exactly, you need to set that to 20MHz when flashing, it has changed, the wiki has not been updated yet.

**Reply by CitizenDev** [reply]
ooh, will try with new build then. Thanks

**Reply by CitizenDev** [reply]
Worked like a charm. Will configure and do some test driving if I can. I'll keep u posted

**Comment by CitizenDev** [comment]
So I drove 5km to work today with new firmware.
Somehow bike is faster than before, this might be due to configuration changes since I did not bring everything over 1:1

But first thing I noticed is that the cut-off window for gear sensor seems to be too short or ramp-down is not fast enough. I got some fairly nasty gear changes within that 5km.
Other thing that I noticed is that the amperage shown on screen seems to be ALOT lower than on previous firmware meanwhile as stated the motor itself seems to give out more power. I did not see over 11A reading once even when riding 50kmh upwind on max gear. Previously this scenario was mostly 20+ amps.

Is it possible that more than rated 25 amps is delivered to the motor if these readings are incorrect?

I will try to further investigate this. 5km is fairly short distance and I didn't have my eyes on the screen the whole time.

**Reply by danielnilsson9** [reply]
Thank you for testing!

Yes, I do not have a gear sensor on my bike so I haven't been able to tune that

The ramp period is in app.c

```
// How long the power interrupt will last when gear sensor is triggered.
#define SHIFT_SENSOR_INTERRUPT_PERIOD_MS		300
```

Please try to experiment with that and see if it helps, it ramps down to 50% of target current.


Looks like there is an error on line 487 in extcom.c:

`uint8_t amp_x2 = (uint8_t)(motor_get_battery_current_x10() * 2) / 10;`

That would mean the reported amps can never be over 12 since the cast to uint8 is done before the division...

Interesting about the amperage I will investigate if the motor control MCU has a different range than the BBSHD, I didn't check that, just assumed it was the same. Please check you configuration too, hitting 50kmh on 25A sounds pretty much spot on though.

**Reply by danielnilsson9** [reply]
Seems like you are right, there is a different range on the BBS02 for current measurements, the current is 23% higher for some strange reason. So yeah, be careful, that means you were pushing 30A...

**Reply by CitizenDev** [reply]
I took a longer way back. I tried really hard and I think maximum that I was was 12.5A. 

I think the overall aggressiveness is most likely due to configuration. Maybe because I did not set speed limit on PAS levels, but set lower target power % than I had originally

In original configuration I also had these to limit heat issues on controller
Start Current 10%
Slow-Start Mode 4
Keep Current 80%
And speed limited to 40kmh (I still felt quite a bit more power at lower speeds though)

I'm not yet completely sure how these settings would look like in your configurator


I will try to fiddle with shift ramp period. Can I also change the 50% ramp down to something even lower?

**Reply by CitizenDev** [reply]
> Seems like you are right, there is a different range on the BBS02 for current measurements, the current is 23% higher for some strange reason. So yeah, be careful, that means you were pushing 30A...

Oooh, so that's why it was flying haha.
It's good then that it was quite cold outside. I used over 85% current PAS mostly on few occasions to try to get higher amp reading.

**Reply by danielnilsson9** [reply]
In this firmware:

Start current is always 0%
Slow-Start is a current ramp, you can configure that in (Current Ramp (A/s)) in the config tool, lower value -> lower acceleration.
Keep current setting does not exist, you will continue to receive maximum assist level power even at you target pedaling cadence for the assist level, should probably add that option if people find it useful.

Yes you can change blow 50% in the code at line 664:
```
// keep power at 1/2 requested
*target_current = (uint8_t)(*target_current / 2u);
```

I have fixed the amp reading, 12.5A matches exactly with the error i found.

**Reply by danielnilsson9** [reply]
I have adjusted the adc conversion ratios for BBS02, both the displayed current reading and motor power configuration should be correct now.

**Reply by CitizenDev** [reply]
Thanks, will build and flash new one as soon as I can

**Reply by CitizenDev** [reply]
Today my bike felt slower (sad emoji) and seemed to display correct values. So seems to be working.
I also doubled the SHIFT_SENSOR_INTERRUPT_PERIOD_MS to 600 and it seemed to help. But I cannot be sure if it was because of overall power decrease or this setting tbh. I did not change target current this time yet. I will be experimenting different combinations in the future.

Would it be possible to make SHIFT_SENSOR_INTERRUPT_PERIOD_MS configurable in Config tool in the future? This setting might vary from bike to bike depending on components like shifters, derailleur, etc.

**Reply by danielnilsson9** [reply]
I changed the default to 600ms, will see if anyone finds that bad. A config could be implemented but I would ideally want to avoid that and not have a million options in the config tool :)

**Comment by CitizenDev** [comment]
As a fun sidenote I think I broke my freewheel yesterday when probably riding at 30A, whoops. The chain keeps turning when coasting without pedalling about half the time.

**Reply by danielnilsson9** [reply]
Thats unfortunate. That's the freewheel on you back wheel hub and not in the motor, right? Very strange, I have been pushing 45A 52V for years without issue.

**Reply by CitizenDev** [reply]
yea the back wheel hub, it gets stuck or something now sometimes and keeps chain turning while coasting. Not a huge issue, but makes coasting less effective so will definitely replace at some point

**Comment by silenec** [comment]
So how is everyone's experience with this? I would test it but is it safe to flash? Is it possible to go back to stock FW?

Some say that the stock FWs floating around are locked down and while you can change the programming settings and save them, they do nothing and the controller always runs with stock settings. Is it true? Do we have access to stock unlocked firmware?

From comments on the Biktrix youtube video (flashing stock FW):

> "This broke my bike, I just get the error 30 now and no power from the motor anymore, anyone know how to help?"

> "warning don't put this on your bike you will ruin your bike"

> "Help! After doing this successfully. I used the Bafang Config Tool software, but if I change even one parameter I get a "30H" error on the display. To cure that I have to re-flash as in the video above. What gives?"

> "Bafang has been changing certain components on the controller recently which means that this firmware may not work."

> "I successfully reprogrammed my motor following these steps however now when I turn on my bike I get an error 30 message. any idea why? It was working fine before reprogramming."

> "I had this same thing happen yesterday, I'm guessing something got corrupted during the update. Have a replacement controller coming. Will update if that helps"

Now I am not too sure. How hard is it to brick the controller if I follow the instructions? Can I always reflash BBS-FW if stock FW does not work? 

My controller says:

Manufacturer: HZXT
Model: SZZ9
Hardware Version: 1.1
Firmware Version : V2.0.1.1
Nominal Voltage: 48V
Max Current: 25A

**Reply by danielnilsson9** [reply]
There is a known issue on the BBS02 version of this firmware causing crash/reset sometimes, so consider that before flashing. https://github.com/danielnilsson9/bbs-fw/issues/56

I did test the linked original firmware on my controller and could change parameters without getting error 30 and they seemed to be applied but now I'm a bit unsure given all the comment from the youtube video, maybe they have reuploaded a newer version, I don't know to be honest.

**Reply by silenec** [reply]
Ok thanks, I will wait until we figure it all out then, just to be safe. And also for the keep current option. By the way I think there might be another "hidden" function in the stock firmware - chain skip / tire slip detection. When that happens, motor stops and it starts to spin up again from 0rpm. I had it happen when I put new chain on worn cassette and also on snow/ice. I only saw it when using PAS, not throttle. Not sure what triggers it exactly and how important or useful it actually is.

**Reply by skrohmi** [reply]
Hello, I flashed my motor succesfully. Didn't get any alerts. I can program the software and it works like it should. Only the display has some issues now: It doesn't show the speed, and power (Watt or Amps) is missing. And I get "Error E30". The speed sensor itself is working, because 25km/h limit works too. Configuration is BBS02, display type 850C and chipset STL 15F2K60S2

**Reply by danielnilsson9** [reply]
Please read the flashing guide carefully, you have to select 20MHz when flashing. https://github.com/danielnilsson9/bbs-fw/wiki/Flashing-the-Firmware

**Comment by skrohmi** [comment]
Hi Daniel, thanks for your fast reply . I did of course use 20MHz like the manual says. Where are you from? If I may ask. Maybe I cab send you a controller. Cheers StefanVon meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: Daniel Nilsson ***@***.***> Datum: 28.09.22  16:51  (GMT+01:00) An: danielnilsson9/bbs-fw ***@***.***> Cc: skrohmi ***@***.***>, Comment ***@***.***> Betreff: Re: [danielnilsson9/bbs-fw] BBS02 (Discussion #48) 
Please read the flashing guide carefully, you have to select 20MHz when flashing. https://github.com/danielnilsson9/bbs-fw/wiki/Flashing-the-Firmware

**Reply by danielnilsson9** [reply]
I see, this is the usual symptom with flashing with the wrong frequency so I was pretty sure it was that. But I may have misread, you are able to connect using the config tool and set parameters? 

If so then I'm afraid you may have broken the display connector when unplugging and replugging, they are quite bad and very sensitive.

**Reply by danielnilsson9** [reply]
I'm from Sweden, but I do not really have any need for controllers right now unless it's for BBS01 or the ULTRA.

Thanks anyways!

**Reply by danielnilsson9** [reply]
To confirm if the display cable broke you could reflash the standard firmware again or try to connect another display if you have one.

**Comment by skrohmi** [comment]
Von meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: Daniel Nilsson ***@***.***> Datum: 28.09.22  16:51  (GMT+01:00) An: danielnilsson9/bbs-fw ***@***.***> Cc: skrohmi ***@***.***>, Comment ***@***.***> Betreff: Re: [danielnilsson9/bbs-fw] BBS02 (Discussion #48) 
Please read the flashing guide carefully, you have to select 20MHz when flashing. https://github.com/danielnilsson9/bbs-fw/wiki/Flashing-the-Firmware

**Reply by skrohmi** [reply]
thanks for helping with a new display works, the displays are the same from the HW and SW so for info, now everything works as it should. I'll send you a fee for the great work

**Reply by danielnilsson9** [reply]
Thank you, that's was generous of you! To anyone reading this, please be careful when unplugging your display and do not pull on the cable, only on the connector. If you pull on the cable it will eventually break.

**Reply by silenec** [reply]
I connect to the main harness connector instead for that reason. That one seems durable. You have to DIY the cable though. https://i.imgur.com/xKQDHIB.jpg

**Comment by skrohmi** [comment]
I fix the problemVon meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: silenec ***@***.***> Datum: 29.09.22  18:04  (GMT+01:00) An: danielnilsson9/bbs-fw ***@***.***> Cc: skrohmi ***@***.***>, Comment ***@***.***> Betreff: Re: [danielnilsson9/bbs-fw] BBS02 (Discussion #48) 
I connect to the main harness connector instead for that reason. That one seems durable. You have to DIY the cable though. https://i.imgur.com/xKQDHIB.jpg

**Comment by kivig2** [comment]
Finally got to testing too. Everything seems to be working properly with no hick-ups. But one thing seems strange - motor engages only after reaching somewhere around 5-8km/h (PAS/variable PAS). Usually that's not a big problem, but is frustrating to start in a deeper snow :)
I glanced over the code and didn't spot any threshold. Could it be having a single magnet? Setting max speed to 100 and actual limit with %?
[bbs02fw-settings.zip](https://github.com/danielnilsson9/bbs-fw/files/10235076/bbs02fw-settings.zip)

**Reply by dav0000000** [reply]
I think your 100ms pas_stop_delay is cadence 25rpm. I set this to 350 which is about 8rpm. You will need to increase pas_start_delay_pulses though, perhaps the default of 5 is good. You have 1, otherwise it will start too quick when just moving the pedals and could be a little dangerous. That could be your problem.

**Reply by kivig2** [reply]
Oh, so I got this wrong:
"Stop delay in milliseconds for when PAS shall disengage when no pedaling is detected."
I somehow read it as a timeout to wait after the pedalling has stopped not the time used to detect no pedaling :)
Thanks!
As for short start - with Throttle modulated PAS that is not a problem as you know what you want when you press throttle. This check probably could be skipped when the throttle gets pressed.

**Reply by dav0000000** [reply]
Yes the pas_stop_delay is the amount of time between receiving a pulse from the pas sensor. If it's 100ms between pulses then that is 2.4s for 24 pulses and a full revolution calculate to 25rpm. You have to exceed this rpm over the number of start pulses you have set  before the motor will start. You are correct that it is the delay after you stop pedalling too. You may still have other issues as I have not tested what you are doing but this could explain why it taking so long to start.

**Comment by Andy-Kona** [comment]
I’ve just installed firmware 1.2.0 onto a bbs02b 500w 36v 25a controller with MCU STC15FSTC15F2K60S2, I had no problem installing the firmware. Currently the motor is on the bench. I’ve been able to connect to it with the config tool and read and save changes. I have also installed the  52v biktrix firmware from the link above on the same controller, and following their guide I was able to install without a problem, now the controller says it’s a 48v 25a. I have also been able to connect with the bafang config too, change the config, save it and the changes get saved ok, so this firmware is not locked.

I’ve noticed that the throttle on firmware 1.2.0 is quite abrupt, it has a very small range, a bit like a on off switch. Made some changes in the config but no real change to throttle response. Now the biktrix firmware gives a nice smooth throttle response across the range of throttle movement! That’s interesting.

I also have a bbs01b 250w controller, its 36v 20a  it came installed to a 250w motor I purchased several years ago. I’ve checked the MCU and its STC15FSTC15F2K60S2 so this should program ok! I don’t want to go any further as I don’t have a way back and want to keep this as a stock 250w controller, but its interesting to see that this controller looks to be compatible. 

I’m planning to install the motor with 1.2.0 firmware to my second bike and do some road tests and will report back.

**Comment by cubebafang** [comment]
Hi,

Firmware is great. thx a lot.

I wondering what is "keep current" value in versio 1.20 ? In orginal firmware it is 60%, which is nice.
Do you have any schedule when you implement "keep current% option"  ?
Could explain for me what mean  "start delay i degrees" and how current ramp actually working ? I know how orginal current ramp working.

https://electricbikereview.com/forums/threads/bbs02b-my-look-at-driver-settings.47120/

https://images89.fotosik.pl/567/996a135434fc1324.jpg

https://images89.fotosik.pl/567/bdffdec61c55c1a5.png

**Reply by danielnilsson9** [reply]
Keep current is not implemented in version 1.2.0, will be part of version 1.3 which will be released when the weather is better and I can do some testing.

Start degrees is how many degrees of a full revolution (360 degree) you need to pedal before PAS kicks in, same as in original firmware.
Current ramp is linear, keep current decreases current linearly with higher cadence.

**Reply by cubebafang** [reply]
Hi, 
Ok i see!

I tested yesterday you ver 1.2 firmware and it was more powerful with same cadence values as orginal firmware ! That's why I asked about "keep current" values

-start degrees is same almost as "crank trigger pulses" on bafang programmer?
-basicly "keep current" is in ver 1.2 100% from the set value, no drops ?
-linear current ramp at 15A/sec. (default) until the set value is reached?

Lets wait version 1.3.  -> i will donate if "keep current" works well 👍 

im from Finland (Uleåborg) here is the best weather of winter season for MTB :)
If you need help for testing "keep current", maybe i can help you. 

you probably knew but the "keep current" option needs some "current decay values" (or something like that) 
see below pic from the original current ramp.

![kuva](https://user-images.githubusercontent.com/123980511/215688585-8470b25a-ab5d-456c-b972-17f4e04522ab.png)

**Reply by danielnilsson9** [reply]
- Yes, start degree is the same as "Start Degree (Signal No). But in degrees instead, there are 24 pulses per pedal revolution, so 4 pulses = 60 degrees.
- Yep thats correct! The current you have set for your assist level is the the one it will try to keep in 1.2.0, which would equal 100% keep current in original firmware I suspect.
- Interesting graph, I have never seen that one before. So from that I would say this firmware works in the following way:

1. Startup degree is the same but in degrees instead of pulses.
2. Slow start mode is called current ramp in this firmware and is set in A/s instead of a number (and is always linear)
3. Current decay parameter doesn't really exist, the current will decrease linearly toward target (which is Keep Current (%) * Assist level target current). Linear decrease will start once Keep Current Cadence configuration parameter is hit.
4. Stop delay is a bit weird, I'm not sure from what point it is counted in the original firmware. In this firmware it is the time between PAS sensor pulses, which means you cannot really set it below 200ms since then it will limit your minimum cadence you have to pedal to get pedal assist. I would recommend to just keep the default setting in this firmware, it is not possible to set it much lower.
5. Stop decay is not configurable in this firmware, it is linear and as fast as possible but slow enough to not cause a jerk when coming from high power to a stop.

If you want, yes please try to build and run master, it should be stable and has the keep current options.
https://github.com/danielnilsson9/bbs-fw/wiki/Build-from-Source

I'm from southen Sweden, it is raining constantly during the winter... So the nice bike is hardly used during this period (a part from a few dry days). I am however running this firmware on my commuting bike with a TSDZ2 instead of a BBSHD and it's working nicely, but I want to do some testing on the BBSHD before release. I will try to make a release if I can do some final testing some weekend. There is no more new functionality planned for 1.3, so the code in in master is probably pretty close to final.

**Reply by cubebafang** [reply]
Hi,

i just run second test on ver 1.2.

It's seems start up assist level does not work. Level is anyway 0 after restart. I tried set  level 3 or 4 but no matter.

I have 750W bbs02b and 840 Wh battery. max. current is set a 20A. Power consumption was with orginal fw on snow 1V@9 km and with ver 1,2 is it on snow 1V@ 6,3 km (with approximately same assist levels, 30-40 % of max. current).
Current consumption is agree with orginal fw because i was keep current 60%. 

I will try to build, it's been a long time when i made last build about anything. more than 25 years with C and about 10 years with LabView

txh.

**Reply by cubebafang** [reply]
Hi again.

I just realized that the "keep current" option would be nice if a value could be set for each assist level. If the assis level is low, the "keep current" value is better to be higher.

**Comment by e-i-k-e** [comment]
Hello,
before I use this firmware. I would like to save the original. Is this possible?

Can you already estimate when you will make version 1.3 available for the bbs02?
I can't do the "build from source" at the moment.

**Comment by e-i-k-e** [comment]
@danielnilsson9 
Do you already know when you will release the next version?
Best regards
